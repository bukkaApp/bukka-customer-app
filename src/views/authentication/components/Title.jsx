import React from 'react';

import PropTypes from 'prop-types';

const Title = ({ slideToNextInput, userEmail }) => {
  if (slideToNextInput) {
    return (
      <div className="form-title mb-4">
        <span>Welcome back,<br /></span>
        <span className="auth-form-title">{userEmail}</span>
      </div>
    );
  }
  return (
    <div className="form-title mb-4">
      <span>Type your email</span>
    </div>
  );
};

export default Title;

Title.defaultProps = {
  slideToNextInput: false,
  userEmail: ''
};

Title.propTypes = {
  userEmail: PropTypes.string,
  slideToNextInput: PropTypes.bool
};
