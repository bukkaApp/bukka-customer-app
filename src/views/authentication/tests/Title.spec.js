import React from 'react';
import Title from 'src/components/Title';


describe('Title component', () => {
  const props = {
    userEmail: 'myemail@email.com',
    slideToNextInput: false,
  };
  const { container, rerender } = render(<Title {...props} />);

  it('renders properly', () => {
    expect(container).toMatchSnapshot();
  });

  it('re renders properly', () => {
    const newProps = {
      ...props,
      slideToNextInput: true
    };
    rerender(<Title {...newProps} />);
    expect(container).toMatchSnapshot();
  });
});
