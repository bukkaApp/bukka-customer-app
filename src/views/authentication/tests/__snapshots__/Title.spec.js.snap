// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component re renders properly 1`] = `
<div>
  <div
    class="form-title mb-4"
  >
    <span>
      Welcome back,
      <br />
    </span>
    <span
      class="auth-form-title"
    >
      myemail@email.com
    </span>
  </div>
</div>
`;

exports[`Title component renders properly 1`] = `
<div>
  <div
    class="form-title mb-4"
  >
    <span>
      Type your email
    </span>
  </div>
</div>
`;
